csharp/ql/test/TestUtilities/InlineExpectationsTest.qll

```diff
@@ -239,12 +239,20 @@ private string getColumnString(TColumn column) {
 
 /**
  * RegEx pattern to match a single expected result, not including the leading `$`. It consists of one or
- * more comma-separated tags containing only letters, digits, `-` and `_` (note that the first character
- * must not be a digit), optionally followed by `=` and the expected value.
+ * more comma-separated tags optionally followed by `=` and the expected value.
+ *
+ * Tags must be only letters, digits, `-` and `_` (note that the first character
+ * must not be a digit), but can contain anything enclosed in a square brackets.
+ *
+ * Examples:
+ * - `tag`
+ * - `tag=value`
+ * - `tag,tag2=value`
+ * - `tag[foo bar]=value`
  */
 private string expectationPattern() {
   exists(string tag, string tags, string value |
-    tag = "[A-Za-z-_][A-Za-z-_0-9]*" and
+    tag = "[A-Za-z-_](?:[A-Za-z-_0-9]|\\[[^\\]]*\\])*" and
```

Comment:
```java
\\[[^\\]]*\\])*
```
If I managed to parse this correctly, it means that you match anything enclosed in [X], but where X can't contain "]", but the regexp matches [[[foo bar]. Maybe consider avoiding the use of "[" in X as well?

Reply:
I think that's a good suggestion, so I implemented this :+1: